### PR TITLE
 GCE/Windows: add a 1.14 (k8sbeta) job

### PR DESCRIPTION
**Pull request by yujuhong**

### CODE REVIEW

File: testgrid/config.yaml [hunk 1]
```diff
@@ -4655,6 +4657,8 @@ dashboards:
     test_group_name: ci-kubernetes-e2e-windows-gce-poc
   - name: windows-gce
     test_group_name: ci-kubernetes-e2e-windows-gce
+  - name: windows-gce-1.4
```
Comment (neolit123) [inline testgrid/config.yaml]: i think should have been 1.14?

Comment (yujuhong) [reply testgrid/config.yaml]: Yep you're right. Sent #11700

**Pull request merged**